janus-idp/operator
# Do not make creating a secret for PostgreSQL a prerequisite

## Description
Changes:
1) Check the backstage deployment and statefulset configuration, if the main container has an envFrom entry
pointing to the secret reference with special name '{POSTGRESQL_SECRET}', a secret is auto generated, and the envFrom entry is updated to backstage-psql-secret-<CR name>. Otherwise, we assume that the user has an existing secret provided and it will not be auto generated.
2) Add a new field existingDbSecret to the backstage CRD to explicitly indicate an existing database secret is used (this is the same way as the helm chart does).

## Which issue(s) does this PR fix or relate to

- Fixes #63 

## PR acceptance criteria

- [X ] Tests

## How to test changes / Special notes to the reviewer
a) Deploy the operator and the dummy backstage CR (examples/bs1.yaml).
b) Check that the psql db and backstage are running.
c) Check that a secret backstage-psql-secret-<CR name> has been created.
d) Delete the backstage CR and check and confirm that the secret gets deleted.
e)deploy examples/bs-existing-secret.yaml and check the backstage and plsql db are using the existing secret used without generating a new one.
